### PR TITLE
State management

### PR DESCRIPTION
**Pull request by O-Reed**
This PR introduces functionality to save and delete state data in a React application.

Save: Allows users to save form input into the application state or a backend API.
Delete: Removes specific state entries based on user actions.
Added error handling for edge cases (e.g., attempting to delete non-existent state entries).
Updated unit tests to ensure coverage for the new save and delete logic.

### CODE REVIEW

File: src/components/UserDetails.jsx [hunk 1]
```diff
@@ -1,28 +1,94 @@
...
+            value={email}
+            onChange={(e) => {
+              setEmail(e.target.value);
+            }}
+          />
+        ) : (
+          <span>{user.email}</span>
+        )}
+        <br />
+      </div>
```
Comment (O-Reed) [inline src/components/UserDetails.jsx:80]: Functionality:

"The save and delete functionalities work as intended, and the logic is simple and readable. Great job on this!"
"Consider adding more robust validation for handleSave to avoid potential edge cases (e.g., duplicate entries)."
Performance:

"The use of filter for deleting state items is appropriate, but it might become less performant if the state grows very large. Consider using an ID for quick lookups in the future."
State Management:

"The implementation relies on local state. If this needs to scale or persist, you might consider integrating a global state management library (e.g., Redux or Zustand)."
Code Style and Best Practices
"The code is well-structured and adheres to React best practices. Nice use of functional updates with setData."
"Consider using constants or enums for button labels to improve maintainability."
Error Handling:
"Good job preventing empty values from being saved. Consider extending error handling to display a user-friendly error message instead of using alert()."
"For handleDelete, you might want to add a confirmation dialog to prevent accidental deletions."
Testing:
"It would be great to see unit tests verifying that:
Items are correctly added to the state.
Items are deleted from the state as expected.
Edge cases like saving empty values are handled properly."

Comment (O-Reed) [reply src/components/UserDetails.jsx:80]: Inline comments provide specific insights into the code.

**Review by O-Reed — COMMENTED** [review]
Great work on implementing the save and delete functionality! The code is clean, readable, and works as intended.
I left a few suggestions regarding error handling, performance optimizations, and testing. Addressing those would make the implementation more robust and scalable. Once updated, I believe this PR is ready to merge. Keep up the excellent work!
